deltacat/compute/compactor/steps/materialize.py

```diff
@@ -36,6 +36,9 @@
 )
 from deltacat.utils.metrics import emit_timer_metrics, MetricsConfig
 
+if importlib.util.find_spec("memray"):
+    import memray
+
```

Comment:
Minor: we may be able to avoid repeating this if statement by creating a separate module. 

```python
# profiling_util.py

import importlib

memray = None
if (importlib.util.find_spec("memray")):
    import memray as mem
    memray = mem
```

and then in materialize or any other file:

```python
from profiling_util import memray
```